### DISCUSSION TITLE
"fetch" in dev version

### DISCUSSION THREAD

**Discussion by bettywolfy** [nesbox/TIC-80#2192] [Ideas]
I think there was a discussion on github "internet function in tic80"

But, I had a more specific idea, a "simpler" way for tic80 to communicate with the internet 

```lua
fetch(url)
fetch(url, options)
```

I think there is a "fetch.lua" library somewhere.

An example of how it could be used:

```lua
function BOOT()
    URL = "https://api.sendmeaquote.com/end"

    quote = fetch(URL, {
        category = "general"
    })
    
    t = 0
end

function TIC()
    cls(0)
    
    printcc("Bounce 'n Bounce", 120, 34, 12, 2)
    printcc((t // 100) .. "% - loading assets...", 120, 68, 13, 1)
    
    printcc("\" " .. quote.content .. " \"", 120, 102, 12, 1)
    printcc("- " .. quote.author, 120, 119, 13, 1)
    
    t = t + 1
end

function printc(text, x, y, color, scale)
    local w = print(text, 512, 512, 0, true, scale, true)
    return print(text, x - (w / 2), y, color, true, scale, true)
end

function printcc(text, x, y, color, scale)
    return printc(text, x, y - (3 * scale), color, scale)
end
```

And here's the code being run with a handled implementation of the fetch function:
![Screenshot_20230511-162334_TIC-80](https://github.com/nesbox/TIC-80/assets/92587199/8da4ef1f-a06d-40b9-a237-e4782aca32f9)

I would like to see the demoscenes that the community would make using this function!

**Comment by bettywolfy** [comment]
Off-Topic: I don't know if this exists, but is it possible to save the tic80 version value in the .tic?

Perhaps for some "This game was made in a newer version of tic80..."

**Comment by nesbox** [comment]
Fetch is not planned, sorry
